src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -1079,31 +1124,30 @@ def __import_kvset_from_file(client, path, strict, yes):
         if __validate_import_config_setting(config_setting):
             kvset_to_import.append(config_setting)
 
-    if strict or not yes:
-        existing_kvset = __read_kv_from_config_store(client,
-                                                     key=SearchFilterOptions.ANY_KEY,
-                                                     label=SearchFilterOptions.ANY_LABEL)
+    existing_kvset = __read_kv_from_config_store(client,
+                                                 key=SearchFilterOptions.ANY_KEY,
+                                                 label=SearchFilterOptions.ANY_LABEL)
     kvset_to_delete = []
     if strict:
         kvset_to_delete = list(filterfalse(lambda kv: any(kv_import.key == kv.key and kv_import.label == kv.label
                                                           for kv_import in kvset_to_import), existing_kvset))
-    if not yes:
-
-        # When strict mode is not enabled, we don't delete configurations if they are missing from the import file,
-        # so don't need to show them in the diff, so omit them from existing kvset
-        if not strict:
-            existing_kvset = list(filter(lambda kv: any(kv_import.key == kv.key and kv_import.label == kv.label
-                                                        for kv_import in kvset_to_import), existing_kvset))
 
-        existing_kvset_list = __serialize_kv_list_to_comparable_json_list(existing_kvset, ImportExportProfiles.KVSET)
-        kvset_to_import_list = __serialize_kv_list_to_comparable_json_list(kvset_to_import, ImportExportProfiles.KVSET)
+    # When strict mode is not enabled, we don't delete configurations if they are missing from the import file,
+    # so don't need to show them in the diff, so omit them from existing kvset
+    if not strict:
+        existing_kvset = list(filter(lambda kv: any(kv_import.key == kv.key and kv_import.label == kv.label
+                                                    for kv_import in kvset_to_import), existing_kvset))
 
-        logger.warning('\n---------------- KVSet Preview ----------------')
-        changes_detected = __print_preview_json_diff(existing_kvset_list, kvset_to_import_list)
-        if not changes_detected:
-            logger.warning('Target configuration store already contains all configuration settings in source. No changes will be made.')
-            return
+    existing_kvset_list = __serialize_kv_list_to_comparable_json_list(existing_kvset, ImportExportProfiles.KVSET)
+    kvset_to_import_list = __serialize_kv_list_to_comparable_json_list(kvset_to_import, ImportExportProfiles.KVSET)
 
+    changes_detected = __print_kvset_json_diff(existing_kvset_list, kvset_to_import_list, yes=yes)
+    
+    if not changes_detected:
+        logger.warning('Target configuration store already contains all configuration settings in source. No changes will be made.')
```

Comment:
Would this laready be logged by __print_kvset_json_diff?

Reply:
My bad! This has been updated. Thanks!